formats/json/commonMain/src/kotlinx/serialization/json/JsonContentPolymorphicSerializer.kt

```diff
@@ -55,11 +55,12 @@ import kotlin.reflect.*
  *
  * // Now both statements will yield different subclasses of Payment:
  *
- * Json.parse(PaymentSerializer, """{"amount":"1.0","date":"03.02.2020"}""")
- * Json.parse(PaymentSerializer, """{"amount":"2.0","date":"03.02.2020","reason":"complaint"}""")
+ * Json.decodeFromString(PaymentSerializer, """{"amount":"1.0","date":"03.02.2020"}""")
+ * Json.decodeFromString(PaymentSerializer, """{"amount":"2.0","date":"03.02.2020","reason":"complaint"}""")
  * ```
  *
- * @param T A root class for all classes that could be possibly encountered during serialization and deserialization.
+ * @param T A root type for all classes that could be possibly encountered during serialization and deserialization.
+ * Must be an interface, abstract or sealed class
```

Comment:
I think we also support `open` classes? Maybe it's better to write just `non-final class or interface`.